polkadot/node/core/provisioner/src/tests.rs

```diff
@@ -780,7 +798,9 @@ mod select_candidates {
 		let cores = [1, 4, 7, 8, 10, 12];
 		let cores_with_code = [1, 4, 8];
 
-		let expected_cores = [1, 7, 10, 12];
+		// We can't be sure which one code upgrade the provisioner will pick. We can only assert
+		// that it only picks one.
+		let expected_cores = [[1, 7, 10, 12], [4, 7, 10, 12], [7, 8, 10, 12]];
```

Comment:
This confused the heck out of me. I kept reading this as cores per para, instead of a listing of all possible outcomes for the cores dropped. Maybe `possible_expected_cores` might be a slightly better name? Honestly not sure it would have helped that much, all clear in hindsight anyways.

Reply:
renamed it to possible_expected_cores as suggested and added some helping comments